numpy/_core/src/umath/ufunc_object.c

```diff
@@ -142,8 +142,10 @@ PyUFunc_clearfperr()
 NPY_NO_EXPORT int
 set_matmul_flags(PyObject *d)
 {
-    PyObject *matmul = _PyDict_GetItemStringWithError(d, "matmul");
-    if (matmul == NULL) {
+    PyObject *matmul = NULL;
```

Comment:
Does this need an init?  If it does, might make sense to mention that in the Python docs, @vstinner, even if not a big deal.  (I.e. that it effectively uses `Py_SETREF(res, ...)` internally.

But this would be a silly nitpick here to actually remove it.

Reply:
You can check the doc: https://docs.python.org/dev/c-api/dict.html#c.PyDict_GetItemRef

Reply:
I tend to initialize variables in C code even if I don't strictly need to. Probably a knee-jerk response from experiences of reading from an uninitialized variable causing heisenbugs.